### PR TITLE
Browser tests

### PR DESCRIPTION
**Pull request by richaagarwal**
Introduces cucumber and an isolated test environment for running browser tests

## Changes

- Modified files to run browser tests in an isolated environment.

## Removals

- Removed page objects which are no longer viable. 

## Testing

- Run `npm install`
- Run `tox -e acceptance`

## TODO

- Create proxy server to blacklist external dependencies.

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by willbarton** [comment]
doge

**Comment by chosak** [comment]
@richaagarwal @sebworks what's the entry point for testing this? I tried `tox -e acceptance` but I get an error like:

```
Exception while running run() in 'scripts.test_data'
...
django.db.utils.OperationalError: no such table: v1_sublandingfilterablepage
```

This script seems to be trying to use a sqlite DB that hasn't been migrated - is there some other step I need to run first?

**Comment by sebworks** [comment]
> This script seems to be trying to use a sqlite DB that hasn't been migrated - is there some other step I need to run first?

No, I'm looking at it now.

File: acceptance-test.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -e
+
+python manage.py migrate
+python manage.py runscript initial_data
+python manage.py runscript test_data
+lsof -i tcp:9500| awk 'NR!=1 {print $2}' | xargs kill
+python manage.py runserver 9500 < /dev/null &
```
Comment (chosak) [inline acceptance-test.sh]: Because this starts the server using `&`, after `tox` finishes, this server is still left running in the background. Can this script (or how `tox` invokes it) be changed to make sure it gets shutdown?

Comment (sebworks) [inline acceptance-test.sh]: yeah, I'm working on that.

**Comment by richaagarwal** [comment]
Sorry this isn't ready for review yet!! My bad. I thought "WIP" captured that.

File: acceptance-test.sh (hunk 1 above)
Comment (sebworks) [inline acceptance-test.sh]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/2893/commits/b827531b3f32ec4b926c108ff17a0db7e5f54ca1

**Comment by willbarton** [comment]
But we're so excited!

**Comment by sebworks** [comment]
ok, you can begin to let your excitement build. I've modified the code to do the following:

- Use [directConnect](https://github.com/angular/protractor/blob/master/docs/server-setup.md#connecting-directly-to-browser-drivers), bypassing the need to run the selenium stand alone server
- Create tox environment to run acceptance tests
- Added test runner which starts server, run `initial_data` process`, and manages in-memory sqllite database
- Modified gulp tasks to pass specs to tox

File: tox.ini [hunk 2]
```diff
@@ -29,6 +29,19 @@ setenv =
...
+[testenv:acceptance]
+recreate = False
+commands =
+     coverage run --source='.' manage.py test {posargs}
+passenv =
+    USER
+setenv =
+    {[testenv]setenv}
+    LANG=en_US.UTF-8
+    LC_ALL=en_US.UTF-8
```
Comment (chosak) [inline tox.ini]: What do these two lines mean?

File: .env_SAMPLE [hunk 3]
```diff
@@ -122,7 +122,6 @@ export WORDPRESS=http://www.consumerfinance.gov
 
 export SITE_DESC=cfgov
 export TEST_HTTP_HOST=localhost
-export TEST_HTTP_PORT=8000
```
Comment (chosak) [inline .env_SAMPLE]: Why is this removed instead of being changed to the recommended 9500?

Comment (richaagarwal) [inline .env_SAMPLE]: The test set-up sets it for you.

File: cfgov/cfgov/settings/test.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(REPOSITORY_ROOT, 'db.sqlite3'),
+        'NAME': os.path.join(REPOSITORY_ROOT, 'test_db.sqlite3'),
```
Comment (richaagarwal) [inline cfgov/cfgov/settings/test.py]: Yeah I think we can change it back. At one point we were going to use it hence the change, but it's no longer needed.
